### PR TITLE
HyperPlay package added and remember me functionality

### PR DESCRIPTION
**Pull request by rob1997**

### CODE REVIEW

File: Packages/io.chainsafe.web3-unity.hyperplay/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "name": "io.chainsafe.web3-unity.hyperplay",
+    "version": "1.0.0",
+    "displayName": "web3.unity SDK HyperPlay",
+    "description": "This package includes the integration of HyperPlay, which you can use with our Gaming SDK.",
+    "license": "LGPL-3.0-only",
+    "licensesUrl": "https://github.com/ChainSafe/web3.unity/blob/main/LICENSE",
+    "documentationUrl": "https://docs.gaming.chainsafe.io/",
+    "dependencies": {
+        "com.unity.nuget.newtonsoft-json": "3.0.2",
+        "io.chainsafe.web3-unity": "2.5.0"
```
Comment (kantagara) [inline Packages/io.chainsafe.web3-unity.hyperplay/package.json:11]: let's up this to 2.6 since final package will be 2.6

Comment (rob1997) [reply Packages/io.chainsafe.web3-unity.hyperplay/package.json]: done

**Review by sneakzttv — APPROVED** [review]

**Review by juans-chainsafe — CHANGES_REQUESTED** [review]
Remember me is working, but when you log out, you can't login again, the issue is that the "Login with Hyperplay" button doesnt do anything.

I see this error log always when I start the session:
```
[Web3] Failed to clear hyperplay-data.json : File not found.
UnityEngine.Debug:LogError (object)
ChainSafe.Gaming.Web3.Unity.UnityLogWriter/<>c__DisplayClass3_0:<LogError>b__0 ()
ChainSafe.Gaming.Evm.Unity.Dispatcher/<ActionWrapper>d__13:MoveNext ()
UnityEngine.MonoBehaviour:StartCoroutine (System.Collections.IEnumerator)
ChainSafe.Gaming.Evm.Unity.Dispatcher/<>c__DisplayClass8_0:<Enqueue>b__0 ()
ChainSafe.Gaming.Evm.Unity.Dispatcher:Update ()
```

And this when I do the logout:
```
UnexpectedException: DB:File Opened Twice, /Users/juanmanuelspoleti/Library/Application Support/DefaultCompany/UnitySampleProject/db3.box, Threads
iBoxDB.ByteCodes.C277.m523[T] (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.ByteCodes.C277.m523 (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.Boolean flushBeforeRead, System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.ByteCodes.C253..ctor (System.String p1, iBoxDB.LocalServer.DatabaseConfig p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.ByteCodes.C252.m196 (iBoxDB.LocalServer.DatabaseConfig p1, System.String p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.ByteCodes.C30..ctor (System.Int64 p1, System.String p2, iBoxDB.LocalServer.DatabaseConfig p3, iBoxDB.LocalServer.Replication.IBoxRecycler p4) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.String name, System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
iBoxDB.LocalServer.DB.Open (System.Int64 destAddr) (at <4b0517faac254d598dd4ca5cab979027>:0)
Plugins.CountlySDK.Helpers.StorageAndMigrationHelper.OpenDB () (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Helpers/StorageAndMigrationHelper.cs:96)
Plugins.CountlySDK.Countly.Init (Plugins.CountlySDK.Models.CountlyConfiguration configuration) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Countly.cs:316)
CountlyAnalytics..ctor (ChainSafe.Gaming.Web3.IChainConfig chainConfig, ChainSafe.Gaming.Web3.IProjectConfig projectConfig) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/CountlyAnalytics.cs:33)
System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) (at <e307bbb467104258887a104f6151f183>:0)
```

**Comment by rob1997** [comment]
> Remember me is working, but when you log out, you can't login again, the issue is that the "Login with Hyperplay" button doesnt do anything.
> 
> I see this error log always when I start the session:
> 
> ```
> [Web3] Failed to clear hyperplay-data.json : File not found.
> UnityEngine.Debug:LogError (object)
> ChainSafe.Gaming.Web3.Unity.UnityLogWriter/<>c__DisplayClass3_0:<LogError>b__0 ()
> ChainSafe.Gaming.Evm.Unity.Dispatcher/<ActionWrapper>d__13:MoveNext ()
> UnityEngine.MonoBehaviour:StartCoroutine (System.Collections.IEnumerator)
> ChainSafe.Gaming.Evm.Unity.Dispatcher/<>c__DisplayClass8_0:<Enqueue>b__0 ()
> ChainSafe.Gaming.Evm.Unity.Dispatcher:Update ()
> ```
> 
> And this when I do the logout:
> 
> ```
> UnexpectedException: DB:File Opened Twice, /Users/juanmanuelspoleti/Library/Application Support/DefaultCompany/UnitySampleProject/db3.box, Threads
> iBoxDB.ByteCodes.C277.m523[T] (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C277.m523 (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.Boolean flushBeforeRead, System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C253..ctor (System.String p1, iBoxDB.LocalServer.DatabaseConfig p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C252.m196 (iBoxDB.LocalServer.DatabaseConfig p1, System.String p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C30..ctor (System.Int64 p1, System.String p2, iBoxDB.LocalServer.DatabaseConfig p3, iBoxDB.LocalServer.Replication.IBoxRecycler p4) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.String name, System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.DB.Open (System.Int64 destAddr) (at <4b0517faac254d598dd4ca5cab979027>:0)
> Plugins.CountlySDK.Helpers.StorageAndMigrationHelper.OpenDB () (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Helpers/StorageAndMigrationHelper.cs:96)
> Plugins.CountlySDK.Countly.Init (Plugins.CountlySDK.Models.CountlyConfiguration configuration) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Countly.cs:316)
> CountlyAnalytics..ctor (ChainSafe.Gaming.Web3.IChainConfig chainConfig, ChainSafe.Gaming.Web3.IProjectConfig projectConfig) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/CountlyAnalytics.cs:33)
> System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) (at <e307bbb467104258887a104f6151f183>:0)
> ```

This is a Countly issue, and it happens when we try and initialize Countly again

**Comment by rob1997** [comment]
> > Remember me is working, but when you log out, you can't login again, the issue is that the "Login with Hyperplay" button doesnt do anything.
> > I see this error log always when I start the session:
> > ```
> > [Web3] Failed to clear hyperplay-data.json : File not found.
> > UnityEngine.Debug:LogError (object)
> > ChainSafe.Gaming.Web3.Unity.UnityLogWriter/<>c__DisplayClass3_0:<LogError>b__0 ()
> > ChainSafe.Gaming.Evm.Unity.Dispatcher/<ActionWrapper>d__13:MoveNext ()
> > UnityEngine.MonoBehaviour:StartCoroutine (System.Collections.IEnumerator)
> > ChainSafe.Gaming.Evm.Unity.Dispatcher/<>c__DisplayClass8_0:<Enqueue>b__0 ()
> > ChainSafe.Gaming.Evm.Unity.Dispatcher:Update ()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And this when I do the logout:
> > ```
> > UnexpectedException: DB:File Opened Twice, /Users/juanmanuelspoleti/Library/Application Support/DefaultCompany/UnitySampleProject/db3.box, Threads
> > iBoxDB.ByteCodes.C277.m523[T] (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.ByteCodes.C277.m523 (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.Boolean flushBeforeRead, System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.ByteCodes.C253..ctor (System.String p1, iBoxDB.LocalServer.DatabaseConfig p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.ByteCodes.C252.m196 (iBoxDB.LocalServer.DatabaseConfig p1, System.String p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.ByteCodes.C30..ctor (System.Int64 p1, System.String p2, iBoxDB.LocalServer.DatabaseConfig p3, iBoxDB.LocalServer.Replication.IBoxRecycler p4) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.String name, System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > iBoxDB.LocalServer.DB.Open (System.Int64 destAddr) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > Plugins.CountlySDK.Helpers.StorageAndMigrationHelper.OpenDB () (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Helpers/StorageAndMigrationHelper.cs:96)
> > Plugins.CountlySDK.Countly.Init (Plugins.CountlySDK.Models.CountlyConfiguration configuration) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Countly.cs:316)
> > CountlyAnalytics..ctor (ChainSafe.Gaming.Web3.IChainConfig chainConfig, ChainSafe.Gaming.Web3.IProjectConfig projectConfig) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/CountlyAnalytics.cs:33)
> > System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) (at <e307bbb467104258887a104f6151f183>:0)
> > ```
> 
> This is a Countly issue, and it happens when we try and initialize Countly again

@kantagara any thoughts

**Comment by juans-chainsafe** [comment]
> > > Remember me is working, but when you log out, you can't login again, the issue is that the "Login with Hyperplay" button doesnt do anything.
> > > I see this error log always when I start the session:
> > > ```
> > > [Web3] Failed to clear hyperplay-data.json : File not found.
> > > UnityEngine.Debug:LogError (object)
> > > ChainSafe.Gaming.Web3.Unity.UnityLogWriter/<>c__DisplayClass3_0:<LogError>b__0 ()
> > > ChainSafe.Gaming.Evm.Unity.Dispatcher/<ActionWrapper>d__13:MoveNext ()
> > > UnityEngine.MonoBehaviour:StartCoroutine (System.Collections.IEnumerator)
> > > ChainSafe.Gaming.Evm.Unity.Dispatcher/<>c__DisplayClass8_0:<Enqueue>b__0 ()
> > > ChainSafe.Gaming.Evm.Unity.Dispatcher:Update ()
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And this when I do the logout:
> > > ```
> > > UnexpectedException: DB:File Opened Twice, /Users/juanmanuelspoleti/Library/Application Support/DefaultCompany/UnitySampleProject/db3.box, Threads
> > > iBoxDB.ByteCodes.C277.m523[T] (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.ByteCodes.C277.m523 (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.Boolean flushBeforeRead, System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.ByteCodes.C253..ctor (System.String p1, iBoxDB.LocalServer.DatabaseConfig p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.ByteCodes.C252.m196 (iBoxDB.LocalServer.DatabaseConfig p1, System.String p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.ByteCodes.C30..ctor (System.Int64 p1, System.String p2, iBoxDB.LocalServer.DatabaseConfig p3, iBoxDB.LocalServer.Replication.IBoxRecycler p4) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.String name, System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > iBoxDB.LocalServer.DB.Open (System.Int64 destAddr) (at <4b0517faac254d598dd4ca5cab979027>:0)
> > > Plugins.CountlySDK.Helpers.StorageAndMigrationHelper.OpenDB () (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Helpers/StorageAndMigrationHelper.cs:96)
> > > Plugins.CountlySDK.Countly.Init (Plugins.CountlySDK.Models.CountlyConfiguration configuration) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Countly.cs:316)
> > > CountlyAnalytics..ctor (ChainSafe.Gaming.Web3.IChainConfig chainConfig, ChainSafe.Gaming.Web3.IProjectConfig projectConfig) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/CountlyAnalytics.cs:33)
> > > System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) (at <e307bbb467104258887a104f6151f183>:0)
> > > ```
> > 
> > 
> > This is a Countly issue, and it happens when we try and initialize Countly again
> 
> @kantagara any thoughts

Just in case, these log errors are not happening in dev, and logout login again is working in dev

**Comment by rob1997** [comment]
> Remember me is working, but when you log out, you can't login again, the issue is that the "Login with Hyperplay" button doesnt do anything.
> 
> I see this error log always when I start the session:
> 
> ```
> [Web3] Failed to clear hyperplay-data.json : File not found.
> UnityEngine.Debug:LogError (object)
> ChainSafe.Gaming.Web3.Unity.UnityLogWriter/<>c__DisplayClass3_0:<LogError>b__0 ()
> ChainSafe.Gaming.Evm.Unity.Dispatcher/<ActionWrapper>d__13:MoveNext ()
> UnityEngine.MonoBehaviour:StartCoroutine (System.Collections.IEnumerator)
> ChainSafe.Gaming.Evm.Unity.Dispatcher/<>c__DisplayClass8_0:<Enqueue>b__0 ()
> ChainSafe.Gaming.Evm.Unity.Dispatcher:Update ()
> ```
> 
> And this when I do the logout:
> 
> ```
> UnexpectedException: DB:File Opened Twice, /Users/juanmanuelspoleti/Library/Application Support/DefaultCompany/UnitySampleProject/db3.box, Threads
> iBoxDB.ByteCodes.C277.m523[T] (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C277.m523 (System.String p1, System.Exception p2) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.Boolean flushBeforeRead, System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.IO.BoxFileStreamConfig.CreateStream (System.String path, iBoxDB.LocalServer.StreamAccess access) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C253..ctor (System.String p1, iBoxDB.LocalServer.DatabaseConfig p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C252.m196 (iBoxDB.LocalServer.DatabaseConfig p1, System.String p2, iBoxDB.LocalServer.Replication.IBoxRecycler p3) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.ByteCodes.C30..ctor (System.Int64 p1, System.String p2, iBoxDB.LocalServer.DatabaseConfig p3, iBoxDB.LocalServer.Replication.IBoxRecycler p4) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.String name, System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.LocalDatabaseServer.GetInstance (System.Int64 address) (at <4b0517faac254d598dd4ca5cab979027>:0)
> iBoxDB.LocalServer.DB.Open (System.Int64 destAddr) (at <4b0517faac254d598dd4ca5cab979027>:0)
> Plugins.CountlySDK.Helpers.StorageAndMigrationHelper.OpenDB () (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Helpers/StorageAndMigrationHelper.cs:96)
> Plugins.CountlySDK.Countly.Init (Plugins.CountlySDK.Models.CountlyConfiguration configuration) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Plugins/Countly SDK/Plugins/CountlySDK/Countly.cs:316)
> CountlyAnalytics..ctor (ChainSafe.Gaming.Web3.IChainConfig chainConfig, ChainSafe.Gaming.Web3.IProjectConfig projectConfig) (at /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/CountlyAnalytics.cs:33)
> System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) (at <e307bbb467104258887a104f6151f183>:0)
> ```

Nvm it was a duplicate instance issue, fixed now

**Review by juans-chainsafe — APPROVED** [review]
Issues were fixed, no more log errors and you can logout and login again.

**Review by sneakzttv — APPROVED** [review]

**Review by kantagara — APPROVED** [review]

**Review by kantagara — APPROVED** [review]

**Pull request merged**